instructions/humus-common.instructions.md

```diff
@@ -0,0 +1,259 @@
+---
+description: 'Common patterns and best practices for all Humus application types (REST, gRPC, Queue, Job)'
+applyTo: '**/*.go'
+---
+
+# Humus Framework - Common Patterns
+
+This file provides common patterns and best practices applicable to all Humus application types (REST, gRPC, Queue, Job). Copy this file along with your application-type-specific instructions to your repository's `.github/` or `instructions/` directory.
+
+## Overview
+
+Humus is a modular Go framework built on [Bedrock](https://github.com/z5labs/bedrock) for creating production-ready REST APIs, gRPC services, jobs, and queue processors. Every application automatically includes OpenTelemetry instrumentation, health monitoring, and graceful shutdown.
+
+## Configuration
+
+**Always embed the appropriate Config type:**
+```go
+type Config struct {
+    rest.Config `config:",squash"`  // For REST services
+    // OR
+    grpc.Config `config:",squash"`  // For gRPC services
+    // OR
+    humus.Config `config:",squash"` // For Job/Queue services
+    
+    // Service-specific configuration
+    Database struct {
+        URL string `config:"url"`
+    } `config:"database"`
+}
+```
+
+**Implement provider interfaces when needed:**
+```go
+func (c Config) Listener(ctx context.Context) (net.Listener, error) {
+    return net.Listen("tcp", fmt.Sprintf(":%d", c.HTTP.Port))
+}
+```
+
+**Use Go templates in config.yaml:**
+```yaml
+openapi:
+  title: {{env "SERVICE_NAME" | default "My Service"}}
+  version: {{env "VERSION" | default "v1.0.0"}}
+
+database:
+  url: {{env "DATABASE_URL" | default "postgres://localhost/mydb"}}
+```
+
+## Error Handling
```

Comment:
this is advanced and really only used for testing humus itself so it should not be included